crates/libs/windows/src/core/event.rs

```diff
@@ -0,0 +1,254 @@
+use super::*;
+use bindings::*;
+use std::sync::*;
+
+/// A type that you can use to declare and implement an event of a specified delegate type.
+///
+/// The implementation is thread-safe and designed to avoid contention between events being
+/// raised and delegates being added or removed.
+pub struct Event<T: Interface + Clone> {
+    swap: Mutex<()>,
+    change: Mutex<()>,
+    delegates: Array<T>,
+}
+
+impl<T: Interface + Clone> Default for Event<T> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<T: Interface + Clone> Event<T> {
+    /// Creates a new, empty `Event<T>`.
+    pub fn new() -> Self {
+        Self { delegates: Array::new(), swap: Mutex::default(), change: Mutex::default() }
+    }
+    /// Registers a delegate with the event object.
+    pub fn add(&mut self, delegate: &T) -> Result<i64> {
+        let mut _lock_free_drop = Array::new();
+        Ok({
+            let change_lock = self.change.lock().unwrap();
```

Comment:
I'm surprised you don't get a warning here for an unused variable. I think this should be `_change_lock`. Similiarly, I think `_lock_free_drop` shouldn't have the `_` prefix since this convention is used for *unused* bindings not bindings that are of secondary importance like here.

Reply:
> I think _lock_free_drop shouldn't have the _ prefix

If I remove it, compiler complains with `value assigned to `lock_free_drop` is never read`.